x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypeConversion.java

```diff
@@ -157,6 +181,9 @@ private static Conversion conversion(DataType from, DataType to) {
 
     private static Conversion conversionToString(DataType from) {
         if (from == DATETIME) {
```

Comment:
Here `DATETIME` comes before `DATE` and below in the same class is the other way around.